32-bit doesn't compile SharedClassesMetaDataHelper.java

https://ci.eclipse.org/openj9/job/Build_JDK8_x86-32_windows_OMR/264
```
22:02:04  c:\Users\jenkins\workspace\Build_JDK8_x86-32_windows_OMR\openj9\debugtools\DDR_VM\src\com\ibm\j9ddr\vm29\pointer\helper\SharedClassesMetaDataHelper.java:31: error: cannot find symbol
22:02:04  			layer = j9shrOffset.cacheLayer().intValue();
22:02:04  			                   ^
22:02:04    symbol:   method cacheLayer()
22:02:04    location: variable j9shrOffset of type J9ShrOffsetPointer
```

Caused by https://github.com/eclipse/openj9/pull/6883

--- a/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/SharedClassesMetaDataHelper.java
+++ b/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/SharedClassesMetaDataHelper.java
@@ -23,14 +23,30 @@ package com.ibm.j9ddr.vm29.pointer.helper;
 
 import com.ibm.j9ddr.CorruptDataException;
 import com.ibm.j9ddr.vm29.pointer.generated.J9ShrOffsetPointer;
+import com.ibm.j9ddr.vm29.types.UDATA;
+import java.lang.reflect.InvocationTargetException;
 
 public class SharedClassesMetaDataHelper {
 	public static int getCacheLayerFromJ9shrOffset(J9ShrOffsetPointer j9shrOffset) throws CorruptDataException {
 		int layer;
 		try {
-			layer = j9shrOffset.cacheLayer().intValue();
-		} catch (NoSuchFieldError | NoSuchMethodError e) {
+			UDATA layerUDATA = (UDATA) j9shrOffset.getClass().getMethod("cacheLayer").invoke(j9shrOffset);
+			layer = layerUDATA.intValue();
+		} catch (NoSuchMethodException e) {
 			layer = 0;
+		} catch (InvocationTargetException e) {
+			Throwable cause = e.getCause();
+			if (cause instanceof NoSuchFieldError) {
+				layer = 0;
+			} else if (cause instanceof CorruptDataException) {
+				throw (CorruptDataException) cause;
+			} else if (cause instanceof RuntimeException) {
+				throw (RuntimeException) cause;
+			} else {
+				throw new CorruptDataException("Error getting the cache layer from J9ShrOffsetPointer", cause);
+			}
+		} catch (IllegalAccessException e) {
+			throw new RuntimeException("Error getting the cache layer from J9ShrOffsetPointer");
 		}
 		return layer;
 	}

